### PR TITLE
Search improvements

### PR DESCRIPTION
**Pull request by kailasnadh790**
Test URLs:
- Before: https://main--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor
- After: https://feat-search-impl2--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor
- After: https://feat-search-impl2--famous-smoke-cigaradvisor--hlxsites.hlx.page/cigaradvisor/search?s=cigar

### CODE REVIEW

File: cigaradvisor/blocks/search-results/search-results.js [hunk 1]
```diff
@@ -93,21 +14,23 @@ function filterData(fullTerm, data) {
...
-  const articles = [];
-  results.forEach((post) => {
-    const filteredArticles = allArticles.filter((obj) => obj.path === getRelativePath(post.path));
-    articles.push(filteredArticles[0]);
-  });
+async function processSearchResults(articles, wrapper, limit, articlesCount) {
   await renderPage(wrapper, articles, limit, articlesCount);
   const loadingImageContainer = document.querySelector('.loading-image-container');
   const articleListWrapper = document.querySelector('.article-list-wrapper');
+  loadingImageContainer.style.transition = 'opacity 2s';
```
Comment (bstopp) [inline cigaradvisor/blocks/search-results/search-results.js:21]: set this as a CSS declaration. JS should just add/remove a class to the parent element which will activate these

**Review by bstopp — CHANGES_REQUESTED** [review]

File: cigaradvisor/blocks/search-results/search-results.js [hunk 2]
```diff
@@ -93,21 +14,23 @@ function filterData(fullTerm, data) {
...
+  loadingImageContainer.style.opacity = 0.25;
   articleListWrapper.style.transition = 'opacity 2s';
-  articleListWrapper.style.opacity = 1;
+  articleListWrapper.style.opacity = 0.5;
   articleListWrapper.style.display = 'block';
-  loadingImageContainer.style.transition = 'opacity 2s';
-  loadingImageContainer.style.opacity = 0;
-  loadingImageContainer.style.display = 'none';
+
+  setTimeout(
```
Comment (bstopp) [inline cigaradvisor/blocks/search-results/search-results.js:27]: No need to timeout here. as long as all the articles are rendered and added to the dom, we can just apply the css.

File: cigaradvisor/blocks/search-results/search-results.js (hunk 1 above)
Comment (kailasnadh790) [reply cigaradvisor/blocks/search-results/search-results.js]: fixed

File: cigaradvisor/blocks/search-results/search-results.js (hunk 2 above)
Comment (kailasnadh790) [reply cigaradvisor/blocks/search-results/search-results.js]: Fixed. But loading of 8 teasers is almost instantaneous and the fade-out effect won't be visible without timer.

**Pull request merged**